src/plugins/intel_cpu/tests/functional/subgraph_tests/src/matmul_quantized_subgraph.cpp

```diff
@@ -0,0 +1,161 @@
+            auto fq2 = ngraph::builder::makeFakeQuantize(weightsNode, ngPrec, 256, {}, {-1.28f}, {1.27f}, {-1.28f}, {1.27f});
+            auto fc = std::make_shared<ngraph::opset1::MatMul>(fq1, fq2, false, false);
+            fc->get_rt_info() = getCPUInfo();
+            fc->set_friendly_name(nameMatmul);
+            auto biasWeightsNode = ngraph::builder::makeConstant(ngPrec, {}, std::vector<float>{0.0f}, true);
+            matMul = std::make_shared<ngraph::opset1::Add>(fc, biasWeightsNode);
+        } else {
+            auto fq2 = ngraph::builder::makeFakeQuantize(inputParams[0], ngPrec, 256, {}, {-1.28f}, {1.27f}, {-1.28f}, {1.27f});
+            matMul = builder::makeMatMul(fq1, fq2, false, true);
+            matMul->get_rt_info() = getCPUInfo();
+            matMul->set_friendly_name(nameMatmul);
+        }
+        if (outType == ElementType::u8)
+            nodeBeforeConv = ngraph::builder::makeFakeQuantize(matMul, ngPrec, 256, {}, {0.0f}, {2.55f}, {0.0f}, {2.55f});
+        else if (outType == ElementType::i8)
+            nodeBeforeConv = ngraph::builder::makeFakeQuantize(matMul, ngPrec, 256, {}, {-1.28f}, {1.27f}, {-1.28f}, {1.27f});
+        else
+            nodeBeforeConv = matMul;
+
+        // matmul->fq->matmul can cover x8*s8->x8 case
+        auto filterWeightsShape = matMul->get_output_shape(0);
+        auto filterWeightsNode = ngraph::builder::makeConstant(element::f32, filterWeightsShape, std::vector<float>{}, true);
+        auto fq3 = ngraph::builder::makeFakeQuantize(filterWeightsNode, ngPrec, 256, {}, {-1.28f}, {1.27f}, {-1.28f}, {1.27f});
+        // only matmul avx2 support s8*s8 input
+        auto matMul2 = builder::makeMatMul(nodeBeforeConv, fq3, false, false);
+
+        function = makeNgraphFunction(ngPrec, inputParams, matMul2, "MatmulBrgemmInt8");
+    }
+
+    void CheckNode(std::shared_ptr<const ov::Model> function, const std::string& nodeName) {
+        ASSERT_NE(nullptr, function);
+        for (const auto &node : function->get_ops()) {
+            const auto & rtInfo = node->get_rt_info();
+            auto getExecValue = [&rtInfo](const std::string & paramName) -> std::string {
+                auto it = rtInfo.find(paramName);
+                IE_ASSERT(rtInfo.end() != it);
+                return it->second.as<std::string>();
+            };
+            if (node->get_friendly_name() == nodeName) {
+                auto primType = getExecValue(ExecGraphInfoSerialization::IMPL_TYPE);
+                ASSERT_TRUE(primTypeCheck(primType)) << "primType is unexpected: " << primType << " Expected: " << selectedType;
+                ASSERT_EQ(node->get_output_element_type(0), outType);
+                ASSERT_EQ(node->get_input_element_type(0), inType);
+            }
+        }
+    }
+};
+
+TEST_P(MatmulBrgemmInt8Test, CompareWithRefs) {
+    // only cover avx2_vnni
+    if (InferenceEngine::with_cpu_x86_avx512_core() || !InferenceEngine::with_cpu_x86_avx2_vnni())
```

Comment:
This test looks pretty generic and defintetelly useful. Can we extend for avx512/AMX isas as well (maybe in separate PR)?

Reply:
Would be added in the separate PR further.